### PR TITLE
Add option for vertical function argument list

### PR DESCRIPTION
**Pull request by Venti-**
I wanted this feature for my self, so I added it. I enjoy consistency, so sometimes I like to have the arguments formatted in a way that doesn't change when something gets renamed.

While writing the test, I noticed a bug in the comment formatting. Not strictly related to this feature, but it triggers with the test so I included the fix in the same pull request.

### CODE REVIEW

File: tests/source/fn-custom-7.rs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// rustfmt-fn_args_layout: Block
+// rustfmt-fn_args_density: Vertical
+// rustfmt-fn_arg_indent: Tabbed
+// rustfmt-fn_brace_style: AlwaysNextLine
```
Comment (nrc) [inline tests/source/fn-custom-7.rs]: Could you also add tests for `rustfmt-fn_args_density: Vertical`, but without the other options?

**Comment by nrc** [comment]
Looks good, thanks for the PR! r+ with the extra tests.

File: tests/source/fn-custom-7.rs (hunk 1 above)
Comment (Venti-) [inline tests/source/fn-custom-7.rs]: Do you mean a new independent test or replacing this one?

Comment (nrc) [inline tests/source/fn-custom-7.rs]: Another, independent one. (May as well keep this test too).

**Comment by marcusklaas** [comment]
:+1:

**Pull request merged**